### PR TITLE
chore: [RTD-2488] Instrument Apache client

### PR DESCRIPTION
**Pull request by TommasoLencioni**
#### Description

This PR proposes to instrument Apache Client 5 with OpenTelemetry.
#### List of Changes
- change HTTP Client builder to instrumented.

#### How Has This Been Tested?

- [ ] Unit Test Suite
- [x] Integration Test Suite
- [ ] Load Tests

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Chore change (non-breaking change which doesn't provide a direct value to users)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/gov/pagopa/rtd/ms/rtdmsfilereporter/feign/config/HttpClientConfig.java [hunk 1]
```diff
@@ -15,14 +15,18 @@
 import org.apache.hc.core5.http.config.Registry;
 import org.apache.hc.core5.http.config.RegistryBuilder;
 import org.apache.hc.core5.ssl.SSLContexts;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
 @Configuration
```
Comment (and-mora) [inline src/main/java/it/gov/pagopa/rtd/ms/rtdmsfilereporter/feign/config/HttpClientConfig.java:22]:
```suggestion
@Configuration
@RequiredArgsConstructor
```

File: src/main/java/it/gov/pagopa/rtd/ms/rtdmsfilereporter/feign/config/HttpClientConfig.java [hunk 2]
```diff
@@ -15,14 +15,18 @@
...
 public class HttpClientConfig {
 
+  @Autowired
+  private ApacheHttpClient5Configuration apacheHttpClient5Configuration;
```
Comment (and-mora) [inline src/main/java/it/gov/pagopa/rtd/ms/rtdmsfilereporter/feign/config/HttpClientConfig.java:26]:
```suggestion
  private final HttpClientBuilder httpClientBuilder;
```

Comment (and-mora) [reply src/main/java/it/gov/pagopa/rtd/ms/rtdmsfilereporter/feign/config/HttpClientConfig.java:26]: here you inject the instrumented bean you created in the configuration class.

File: src/main/java/it/gov/pagopa/rtd/ms/rtdmsfilereporter/feign/config/HttpClientConfig.java [hunk 3]
```diff
@@ -37,6 +41,7 @@ CloseableHttpClient getHttpClient()
     PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager(
         registry);
 
-    return HttpClients.custom().setConnectionManager(connectionManager).build();
+    return apacheHttpClient5Configuration.createHttpClientBuilder().setConnectionManager(connectionManager)
```
Comment (and-mora) [inline src/main/java/it/gov/pagopa/rtd/ms/rtdmsfilereporter/feign/config/HttpClientConfig.java:44]: here you use the `httpClientBuilder` injected 
```suggestion
    return httpClientBuilder.setConnectionManager(connectionManager)
```

File: src/main/java/it/gov/pagopa/rtd/ms/rtdmsfilereporter/telemetry/AppInsightConfig.java [hunk 4]
```diff
@@ -46,4 +48,10 @@ public MongoClientSettingsBuilderCustomizer mongoOpenTelemetryBridge(
     return clientSettingsBuilder -> clientSettingsBuilder
         .addCommandListener(MongoTelemetry.builder(openTelemetry).build().newCommandListener());
   }
+
+  @Bean
+  public HttpClientBuilder createBuilder(OpenTelemetry openTelemetry) {
```
Comment (and-mora) [inline src/main/java/it/gov/pagopa/rtd/ms/rtdmsfilereporter/telemetry/AppInsightConfig.java:53]: i think this is redundant with the @Bean injected into `ApacheHttpClient5Configuration.java`

**Review by and-mora — CHANGES_REQUESTED** [review]

File: src/main/java/it/gov/pagopa/rtd/ms/rtdmsfilereporter/feign/config/HttpClientConfig.java [hunk 5]
```diff
@@ -15,14 +15,18 @@
...
 import org.apache.hc.core5.http.config.RegistryBuilder;
 import org.apache.hc.core5.ssl.SSLContexts;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
 @Configuration
 public class HttpClientConfig {
 
+  @Autowired
```
Comment (and-mora) [inline src/main/java/it/gov/pagopa/rtd/ms/rtdmsfilereporter/feign/config/HttpClientConfig.java:25]:
```suggestion
```

**Review by and-mora — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 6]
```diff
@@ -86,6 +86,11 @@
 			<artifactId>opentelemetry-logback-appender-1.0</artifactId>
 			<scope>runtime</scope>
 		</dependency>
+		<dependency>
+			<groupId>io.opentelemetry.instrumentation</groupId>
+			<artifactId>opentelemetry-apache-httpclient-5.2</artifactId>
+			<version>2.3.0-alpha</version>
```
Comment (and-mora) [inline pom.xml:92]: this version can be omissed? the bom is already imported

Comment (TommasoLencioni) [reply pom.xml]: Right, thanks!

**Review by lucaconsalvi — APPROVED** [review]

File: src/main/java/it/gov/pagopa/rtd/ms/rtdmsfilereporter/telemetry/AppInsightConfig.java [hunk 7]
```diff
@@ -2,9 +2,11 @@
 
 import com.azure.monitor.opentelemetry.exporter.AzureMonitorExporterBuilder;
 import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.apachehttpclient.v5_2.ApacheHttpClient5Telemetry;
 import io.opentelemetry.instrumentation.mongo.v3_1.MongoTelemetry;
 import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder;
 import io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizerProvider;
+import org.apache.hc.client5.http.impl.classic.HttpClientBuilder;
```
Comment (and-mora) [inline src/main/java/it/gov/pagopa/rtd/ms/rtdmsfilereporter/telemetry/AppInsightConfig.java:9]:
```suggestion
```

File: src/main/java/it/gov/pagopa/rtd/ms/rtdmsfilereporter/telemetry/AppInsightConfig.java [hunk 8]
```diff
@@ -2,9 +2,11 @@
 
 import com.azure.monitor.opentelemetry.exporter.AzureMonitorExporterBuilder;
 import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.apachehttpclient.v5_2.ApacheHttpClient5Telemetry;
```
Comment (and-mora) [inline src/main/java/it/gov/pagopa/rtd/ms/rtdmsfilereporter/telemetry/AppInsightConfig.java:5]:
```suggestion
```

**Review by and-mora — CHANGES_REQUESTED** [review]

File: src/main/java/it/gov/pagopa/rtd/ms/rtdmsfilereporter/feign/StorageAccountRestConnector.java [hunk 9]
```diff
@@ -1,5 +1,6 @@
 package it.gov.pagopa.rtd.ms.rtdmsfilereporter.feign;
 
+import it.gov.pagopa.rtd.ms.rtdmsfilereporter.feign.config.ApacheHttpClient5Configuration;
```
Comment (and-mora) [inline src/main/java/it/gov/pagopa/rtd/ms/rtdmsfilereporter/feign/StorageAccountRestConnector.java:3]:
```suggestion
```

**Review by and-mora — CHANGES_REQUESTED** [review]

**Review by and-mora — APPROVED** [review]

**Pull request merged**